RiotSwiftUI/Modules/Authentication/Registration/AuthenticationRegistrationViewModelProtocol.swift

```diff
@@ -0,0 +1,34 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+protocol AuthenticationRegistrationViewModelProtocol {
+    
+    @MainActor var completion: ((AuthenticationRegistrationViewModelResult) -> Void)? { get set }
```

Comment:
If I understand the flow correct, the `completion` flow can be called multiple types depending on the step (e.g. select server, validate user ...). In that case `completion` might not be the best name, as it suggests either one-time event, or the final event in a series. Could be instead called `resultAction` or something along those lines.

Reply:
Yep, in the Spaces screens I believe Gil renamed this to `callback`. Happy to do match that, although there are view models and coordinators that behave this way so probably better a future PR?

Reply:
Yea that works